server/lib/util.py

```diff
@@ -828,6 +773,28 @@ def _get_highest_coverage_date(observation_entity_counts_by_date,
     max_years_to_check: Only consider entity counts going back this number of
       years
   """
+  recent_date_counts_dict = {}
+  for observation_entity_counts_by_date in observation_dates_by_variable:
+    recent_date_counts = _get_recent_date_counts(
+        observation_entity_counts_by_date, facet_ids, max_dates_to_check,
+        max_years_to_check)
+    for date_count in recent_date_counts:
+      if not date_count["date"] in recent_date_counts_dict:
```

Comment:
nit: could make a temp variable for date_count["date"] since it gets used multiple times

Reply:
done